plugins/aea-ledger-ethereum/aea_ledger_ethereum/ethereum.py

```diff
@@ -50,16 +51,62 @@
 _default_logger = logging.getLogger(__name__)
 
 _ETHEREUM = "ethereum"
-GAS_ID = "gwei"
 TESTNET_NAME = "ganache"
 DEFAULT_ADDRESS = "http://127.0.0.1:8545"
 DEFAULT_CHAIN_ID = 1337
-DEFAULT_GAS_PRICE = "50"
 DEFAULT_CURRENCY_DENOM = "wei"
 _ABI = "abi"
 _BYTECODE = "bytecode"
 
 
+def get_gas_price_strategy(
+    gas_price_strategy: Optional[str] = None, api_key: Optional[str] = None
+) -> Callable:
```

Comment:
can you please provide the expected type of input arguments and the type of the return value, if possible?